File: 05_hash_tables/swift/02_check_voter.swift
import Foundation

var voter = [String : Bool]()

func checkVoter(_ name : String) {
    if voter[name] != nil {
        print("kick them out!")
    } else {
        voter[name] = true
        print("let them vote!")
    }
}

checkVoter("tom")
checkVoter("mike")
checkVoter("mike")

File: 05_hash_tables/elixir/02_check_voter.exs
defmodule Voting do
  def check_voter(voters_map, name) do
    if Map.has_key?(voters_map, name) do
      IO.puts("kick them out!")
    else
      IO.puts("let them vote!")
      Map.put(voters_map, name, true)
    end
  end
end

Voting.check_voter(%{}, "tom")
|> Voting.check_voter("mike")
|> Voting.check_voter("mike")
